### DISCUSSION TITLE
Is the database count as part of the token?

### DISCUSSION THREAD

**Discussion by danieltty** [sinaptik-ai/pandas-ai#781] [Q&A]
If so, it will consume a lot of tokens and cost a lot. But I guess the data analysis part is done on the pandas side, without going into Openai api, right?

**Comment by gventuri** [comment]
Hi @tytung2020 we don't send the whole dataset, but only a few samples and the headers, so only the tokens for these records are counted!

**Reply by danieltty** [reply]
That's great. Even if the user query requires an analysis of most or even all of the rows in the database?
